### PR TITLE
[Linker] Remove bundled mscorlib.xml descriptor from mmp and mtouch

### PR DESCRIPTION
**Pull request by dalexsoto**
This fixes xamarin/xamarin-macios#3749

We currently process [both mscorlib.xml descriptors](https://github.com/xamarin/xamarin-macios/issues/3749#issuecomment-373770268) from the mtouch/mmp
bundle and also the one contained in mscorlib.dll as resource, we are now removing
the descriptor bundled inside mtouch and mmp in favour of the one bundled in
mscorlib.dll.

Source descriptors files for diff:

mono (master): https://github.com/mono/mono/blob/ea4274f2eb74cddf23c9c219b086cd41add0efce/mcs/class/corlib/LinkerDescriptor/mscorlib.xml
XI (master): https://github.com/xamarin/xamarin-macios/blob/da6db5f6601b74fbc8324e974c83a4c33186dc12/tools/linker/Descriptors/mscorlib.xml
XM (master): https://github.com/xamarin/xamarin-macios/blob/da6db5f6601b74fbc8324e974c83a4c33186dc12/tools/mmp/linker/Descriptors/mscorlib.xml

Diffs:

Mono - XI descriptor diff: https://gist.github.com/dalexsoto/f0b1c9c66bf50edf8198063ec039a17b
Mono - XM descriptor diff: https://gist.github.com/dalexsoto/06b253a6743d366a4b9addc21a1e2c2a

### CODE REVIEW

File: tools/mmp/Makefile [hunk 1]
```diff
@@ -104,7 +104,6 @@ tuner_sources = \
 	../../src/build/mac/Constants.cs
 
 linker_resources = \
```
Comment (rolfbjarne) [inline tools/mmp/Makefile]: Can't you remove the entire `linker_resources` variable?

File: tools/mtouch/Makefile [hunk 2]
```diff
@@ -15,7 +15,6 @@ MONOLINKER=$(LINKER_TOOLS_PATH)/linker
 MONO_TUNER=$(LINKER_TOOLS_PATH)/tuner
 
 LINKER_RESOURCES = \
```
Comment (rolfbjarne) [inline tools/mtouch/Makefile]: Same: redundant variable now.

**Comment by dalexsoto** [comment]
@rolfbjarne / @spouliot done! Thanks for the feedback!

**Comment by dalexsoto** [comment]
This PR is blocked on https://github.com/mono/mono/issues/7716

**Comment by dalexsoto** [comment]
Two tests are failing:

`BaseOptimizeGeneratedCodeTest_SetupBlockPerfTest`- Known issue -> https://github.com/xamarin/maccore/issues/649

The second one is `xammac_tests/Release (all optimizations)` which is an issue with the xml descriptor missing this entry `<namespace fullname="System.Security.Cryptography" feature="crypto" />`, that entry seems not ideal so Marek will have a look.

Also this PR can only go in until the descriptor is fixed in mono and mono mono 2018-02  https://github.com/xamarin/xamarin-macios/pull/3402 is merged and rebased with the mono fix.

**Pull request merged**